substrate/frame/revive/README.md

```diff
@@ -0,0 +1,101 @@
+# Revive Module
+
+This module provides functionality for the runtime to deploy and execute PolkaVM smart-contracts.
+
+## Overview
+
+This module extends accounts based on the [`frame_support::traits::fungible`] traits to have smart-contract
+functionality. It can be used with other modules that implement accounts based on [`frame_support::traits::fungible`].
+These "smart-contract accounts" have the ability to instantiate smart-contracts and make calls to other contract and
+non-contract accounts.
+
+The smart-contract code is stored once, and later retrievable via its `code_hash`. This means that multiple
+smart-contracts can be instantiated from the same `code`, without replicating the code each time.
+
+When a smart-contract is called, its associated code is retrieved via the code hash and gets executed. This call can
+alter the storage entries of the smart-contract account, instantiate new smart-contracts, or call other smart-contracts.
+
+Finally, when an account is reaped, its associated code and storage of the smart-contract account will also be deleted.
+
+### Weight
+
+Senders must specify a [`Weight`](https://paritytech.github.io/substrate/master/sp_weights/struct.Weight.html) limit
+with every call, as all instructions invoked by the smart-contract require weight. Unused weight is refunded after the
+call, regardless of the execution outcome.
+
+If the weight limit is reached, then all calls and state changes (including balance transfers) are only reverted at the
+current call's contract level. For example, if contract A calls B and B runs out of weight mid-call, then all of B's
+calls are reverted. Assuming correct error handling by contract A, A's other calls and state changes still persist.
+
+One `ref_time` `Weight` is defined as one picosecond of execution time on the runtime's reference machine.
+
+### Revert Behaviour
+
+Contract call failures are not cascading. When failures occur in a sub-call, they do not "bubble up", and the call will
+only revert at the specific contract level. For example, if contract A calls contract B, and B fails, A can decide how
+to handle that failure, either proceeding or reverting A's changes.
+
+## Interface
+
+### Dispatchable functions
+
+Those are documented in the [reference
+documentation](https://paritytech.github.io/substrate/master/pallet_revive/index.html#dispatchable-functions).
+
+## Usage
+
+This module executes PolkaVM smart contracts. These can potentially be written in any language that compiles to
+Wasm. However, using a language that specifically targets this module will make things a lot easier. Curren
```

Comment:
Maybe it slipped your mind to finish this section? 😅

Reply:
It was late at night :D. Fixed.